sdk/core/Azure.Core/src/AsyncPageable.cs

```diff
@@ -13,6 +13,37 @@ namespace Azure
     /// iterate over.
     /// </summary>
     /// <typeparam name="T">The type of the values.</typeparam>
+    /// <example>
+    /// Example of enumerating an AsyncPageable using the <c> async foreach </c> loop:
+    /// <code snippet="Snippet:AsyncPageable">
```

Comment:
Any idea what will happen if we run this through our ref doc generation tools?  Might want to hold off on merging this until we're through with the release so @scbedd and @chidozieononiwu don't have any fun surprises.

Reply:
I expect something like this https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.windowsservices.webhostwindowsserviceextensions.runasservice?view=aspnetcore-3.0#examples

I'll hold of this PR until the release cycle is done.

Reply:
Sorry, my question was more about whether a custom `snippet` attribute on the `<code>` element would cause any problems with the doc generation tools.  It _shouldn't_, but I've found previous generations of doc generation tools like Sandcastle to be very finicky.  Might be worth an extra doc generation test after we merge it as a sanity check between releases.